src/main/java/duke/DukeCommandMatcher.java

```diff
@@ -0,0 +1,190 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import duke.exceptions.CommandNotFoundException;
+import duke.exceptions.DateFormatException;
+import duke.exceptions.LackOfTimeException;
+import duke.exceptions.NullCommandContentException;
+import duke.exceptions.NullCommandException;
+import duke.exceptions.TaskNotSpecifyException;
+import duke.exceptions.TaskOutOfBoundException;
+import duke.storage.Storage;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.SingletonTaskList;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+import duke.utils.Constants;
+import duke.utils.UtilFunction;
+
+
+/**
+ * The class to match the command to Duke's response.
+ */
+public class DukeCommandMatcher {
+
+    private static final List<String> commandList = new ArrayList<>(Arrays.asList(Constants.LISTPATTERN,
+            Constants.EXITPATTERN, Constants.DONEPATTERN, Constants.TODOPATTERN, Constants.DEADLINEPATTERN,
+            Constants.EVENTPATTERN, Constants.DELETEPATTERN, Constants.FINDPATTERN
+    ));
+
+    private SingletonTaskList taskList;
+
+    public DukeCommandMatcher(Storage database) {
+        this.taskList = SingletonTaskList.getInstance(database);
+    }
+
+    /**
+     * Match the command to corresponding behavior.
+     * @param command
+     * @return message of implementation
+     * @throws CommandNotFoundException
+     * @throws NullCommandException
+     * @throws LackOfTimeException
+     * @throws NullCommandContentException
+     * @throws TaskOutOfBoundException
+     * @throws TaskNotSpecifyException
+     * @throws DateFormatException
```

Comment:
you should specify when these exceptions are thrown!
eg `@throws CommandNotFoundException If the user command is not in the command list`

Reply:
Thanks for the remind, I forgot to leave a readable note for them. Will add soon.